tests/Unit/Metadata/Service/ClassMetadataSearcherTest.php

```diff
@@ -163,6 +163,45 @@ public function testFindAllUsingElasticSearch(): void
         $this->assertSame('class1', $rawResult[0]['class']);
     }
 
+    public function testFindAllUsingElasticSearchWithEmptyClassProperties(): void
+    {
+        $property = $this->createMock(core_kernel_classes_Property::class);
+        $property->method('getRelatedClass')
+            ->willReturn(null);
+
+        $class = $this->createMock(core_kernel_classes_Property::class);
+        $class->method('getLabel')
+            ->willReturn('Class label');
+
+        $this->ontology
+            ->method('getProperty')
+            ->willReturn($property);
+
+        $this->ontology
+            ->method('getClass')
+            ->willReturn($class);
+
+        $this->advancedSearchChecker
+            ->method('isEnabled')
+            ->willReturn(true);
+
+        $this->elasticSearch
+            ->method('search')
+            ->willReturn(new SearchResult([], 0));
+
+        $result = $this->subject->findAll(
+            new ClassMetadataSearchInput(
+                (new ClassMetadataSearchRequest())->setClassUri('class1')
+            )
+        );
+
+        $rawResult = json_decode(json_encode($result->jsonSerialize()), true);
```

Comment:
Why do we have to encode and decode it again?

Reply:
Here I was following the same approach as in tests that were already here for this component (I think this is used to get an anonymous object with all the result properties, so we can test the individual properties with a single `assert()`). By the way, I'll try to change this test to test individual items.